### ISSUE TITLE
Codesniffer says one-line comment for @var, after fix asks back for multi-line

### ISSUE THREAD

**Issue by vojtamares** [slevomat/coding-standard#335]
Hello,

I am writing an tiny library for Elasticsearch and Nette fw and I wanted to write some tests, so I am using the [ninjify/qa](https://github.com/ninjify/qa) (it is using slevomat coding-standard). And after I fix the problem, it says it is a problem again and it wants me to reverse the changes and then again...

Code snippet:
```php
...
class ElasticsearchExtension extends CompilerExtension
{

	/** @var array|string[] */
    public $defaults = [
    		'hosts' => [
    				'localhost',
				],
		];
...
```

But after running codesniffer I get the following error:  Found one-line comment for property \FiolaSoft\Elasticsearch\DI\ElasticsearchExtension::$defaults, use multi-line comment instead.(SlevomatCodingStandard.Commenting.DisallowOneLinePropertyDocComment.OneLinePropertyComment)

But after a change from one-line docs to multi-line I get the same error just for multiline...
Found multi-line comment for property \FiolaSoft\Elasticsearch\DI\ElasticsearchExtension::$defaults with single line content, use one-line comment instead.(SlevomatCodingStandard.Commenting.RequireOneLinePropertyDocComment.MultiLinePropertyComment)

And then the same problem with constant definition TEMP_DIR from [ninjify/nunjuck](https://github.com/ninjify/nunjuck): Constant TEMP_DIR should be referenced via a fully qualified name. (SlevomatCodingStandard.Namespaces.FullyQualifiedGlobalConstants.NonFullyQualified)

Full travis built log: https://travis-ci.org/FiolaSoft/elasticsearch/jobs/370813442#L426

Autofixer offers help, but when I tried to let it fix it, I ended up with an error. It is possible that this bug comes from the [ruleset](https://github.com/ninjify/coding-standard/blob/master/ruleset.xml), but I am not sure about that.

Our repository can be found there: [FiolaSoft/elasticsearch](https://github.com/FiolaSoft/elasticsearch)

**Comment by Majkl578** [comment]
If I understand correctly, you are using https://github.com/ninjify/coding-standard/blob/master/ruleset.xml.
This ruleset seems to import whole SlevomatCodingStandard and selectively exclude some sniffs. This is **not** supported. See https://github.com/slevomat/coding-standard#using-all-sniffs-from-the-standard

This repository is not meant to be used as final CS, rather to selectively enable specific sniffs from here. Some sniffs here are actually mutually exclusive (which is the case you have issues with).

ninjify/coding-standard needs to be fixed to use Slevomat CS correctly - cc @f3l1x

Also I can recommend using i.e. [Doctrine CS](https://github.com/doctrine/coding-standard).

**Comment by f3l1x** [comment]
Yep, you right guys. It needs to be fixed.

**Comment by vojtamares** [comment]
@Majkl578 oh, ok. Well I am new to coding-standard, so I missed that fact quite easily. 

If I want to fix it on my own, I should probably create own CS (FiolaSoft/coding-standard) and fork the ninjify/qa. Or be laizy and let someone else fix itfor me (in ninjify/coding-standard).

But since you recommand using Doctrine CS, I will give it a try on my own! Thank you for quick response!
